pkg/server/plugin/nodeattestor/awsiid/organization_test.go

```diff
@@ -0,0 +1,104 @@
+package awsiid
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testAccountListTTL = "1m"
+)
+
+func TestIsMemberAccount(t *testing.T) {
+	testOrgValidationConfig := &orgValidationConfig{
+		AccountID:      testAccountID,
+		AccountRole:    testProfile,
+		AccountRegion:  testRegion,
+		AccountListTTL: testAccountListTTL,
+	}
+	testOrgValidator := newOrganizationValidationBase(testOrgValidationConfig)
+	err := testOrgValidator.configure(testOrgValidationConfig)
+	require.NoError(t, err)
+
+	testClient := newFakeClient()
+
+	// pass valid account
+	ok, err := testOrgValidator.IsMemberAccount(context.Background(), testClient, testAccountID)
+	require.NoError(t, err)
+	require.Equal(t, ok, true)
+
+	// fail valid account doesnt exist
+	ok, err = testOrgValidator.IsMemberAccount(context.Background(), testClient, "9999999")
+	require.NoError(t, err)
+	require.Equal(t, ok, false)
+}
+
+func TestCheckIfOrgAccountListIsStale(t *testing.T) {
+	testOrgValidationConfig := &orgValidationConfig{
+		AccountID:      testAccountID,
+		AccountRole:    testProfile,
+		AccountRegion:  testRegion,
+		AccountListTTL: testAccountListTTL,
+	}
+	testOrgValidator := newOrganizationValidationBase(testOrgValidationConfig)
+	err := testOrgValidator.configure(testOrgValidationConfig)
+	require.NoError(t, err)
+
+	testIsStale := testOrgValidator.checkIfOrgAccountListIsStale(context.Background())
+	require.Equal(t, testIsStale, true)
+
+	// seed account list and it should return false
+	_, err = testOrgValidator.reloadAccountList(context.Background(), newFakeClient(), false)
+	require.NoError(t, err)
+	testIsStale = testOrgValidator.checkIfOrgAccountListIsStale(context.Background())
+	require.Equal(t, testIsStale, false)
+}
+
+func TestReloadAccountList(t *testing.T) {
+	testOrgValidationConfig := &orgValidationConfig{
+		AccountID:      testAccountID,
+		AccountRole:    testProfile,
+		AccountRegion:  testRegion,
+		AccountListTTL: testAccountListTTL,
+	}
+	testOrgValidator := newOrganizationValidationBase(testOrgValidationConfig)
+	err := testOrgValidator.configure(testOrgValidationConfig)
+	require.NoError(t, err)
+
+	testClient := newFakeClient()
+
+	// check once config is provided correctly and catchburst is false, account list popped up along with timestamp
+	_, err = testOrgValidator.reloadAccountList(context.Background(), testClient, false)
+	require.NoError(t, err)
+	require.Equal(t, len(testOrgValidator.orgListAccountMap), 1)
+	require.Greater(t, testOrgValidator.orgListAccountMapCreationTime, time.Now().Add(-10*time.Second))
```

Comment:
mmm I think you need to use clock here... that can be used to trigger test cases as expected, and you can verify real values